server/src/main/java/org/elasticsearch/action/fieldcaps/TransportFieldCapabilitiesAction.java

```diff
@@ -62,7 +62,7 @@
 
 import static org.elasticsearch.action.search.TransportSearchHelper.checkCCSVersionCompatibility;
 
-public class TransportFieldCapabilitiesAction extends HandledTransportAction<FieldCapabilitiesRequest, FieldCapabilitiesResponse> {
+public class TransportFieldCapabilitiesAction extends RegistrableTransportAction<FieldCapabilitiesRequest, FieldCapabilitiesResponse> {
```

Comment:
This is for the 2nd solution: keep registration code close to the action implementation code (same class)